### ISSUE TITLE
Sanity: "pagination should detect volumes added between pages" removes volumes it didn't create & other issues

### ISSUE THREAD

**Issue by davidz627** [kubernetes-csi/csi-test#223]
This test specifically checks for "existing volumes" by doing a `ListVolumes` call then afterwards does `removing all listed volumes`

This means that this test could removes all volumes listed even if the volume was not created by this test. This could cause data loss depending on how a consumer is using CSI sanity.

This brings up a larger problem I have with this test too - either
1) You are consuming csi-sanity and mocking out the back-end (this is what I do) -> this test is way too complicated and depends on the workings of pagination and doesn't feel like a simple "sanity test" like the rest of the package
2) You are not mocking out the backend -> this test has potential for data loss

**Comment by msau42** [comment]
I agree that the test should only remove volumes created by the test.
/kind bug
/help

**Comment by timoreimann** [comment]
I'd be happy to take a stab at this one.

**Comment by pohly** [comment]
Note that the test is currently disabled (https://github.com/kubernetes-csi/csi-test/pull/226) because it was so problematic.
